### PR TITLE
Suggestions building on top of JOLT

### PR DESCRIPTION
**Pull request by transistive**
Hello @wgevaert,

Thank you for the great PR https://github.com/neo4j-php/neo4j-php-client/pull/123.

I made some changes to keep other objects from becoming polluted with more logic. There is now a legacyformatter and jolt formatter. The ogm formatter decides which one to use based on the version.

All there is left to do is actually formatting the data. I don't think I'll get around to this today, but who knows :shrug: 

Kind regards,

Ghlen

### CODE REVIEW

File: src/Formatter/Specialised/LegacyHttpFormatter.php [hunk 1]
```diff
@@ -195,7 +234,7 @@ private function translateProperties(array $properties): CypherMap
                 $tbr[$key] = $this->translateProperties($castedValue);
             } elseif (is_array($value)) {
                 /** @var array<string, array|stdClass|scalar|null> $value */
-                $tbr[$key] = new CypherList($this->translateProperties($value)->values());
+                $tbr[$key] = new CypherList($this->translateProperties($value));
```
Comment (wgevaert) [inline src/Formatter/Specialised/LegacyHttpFormatter.php:237]: Is this change intentional? It seems a bit unrelated to everything else and I don't know the package well enough to know why this happened

Comment (transistive) [reply src/Formatter/Specialised/LegacyHttpFormatter.php:237]: Yes, it was intentional. Values triggers evaluation and thus mutation. If we just pass the iterable, it will lazily evaluate.

**Comment by wgevaert** [comment]
Hello Ghlen,

Thank you very much for the pull request, this code certainly is much cleaner! I'll merge later today.
I have one remark, but I think it won't mean much.

Cheers!
Wgevaert
